artemis-core-client/src/main/java/org/apache/activemq/artemis/core/message/impl/CoreMessage.java

```diff
@@ -578,34 +567,41 @@ public CoreMessage setUserID(UUID userID) {
    /**
     * I am keeping this synchronized as the decode of the Properties is lazy
     */
-   protected TypedProperties checkProperties() {
+   protected final TypedProperties getOrCreateProperties() {
```

Comment:
Just name this getProperties. The name otherwise is too confusing with your extracted method

Reply:
All it does it invoke / delegate this method, as such if anything i see naming CoreMessage.getProperties cleaner. And there is no need to change it from final, simply you remove the impl in ClientMessageInternalImpl as it would just inherit.

```
   public TypedProperties getProperties() {
      return this.checkProperties();
   }
```

Reply:
@franz1981 did you make the change?

Reply:
Yes, just pushed, but I haven't had the chance yet to ask @mtaylor or @clebertsuconic on how to remove the cleanup internal properties!

Reply:
Nudge :) keen this doesnt lose traction